lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -379,7 +385,7 @@ export class UserAgentApplication {
         this.clientId, scopes,
         ResponseTypes.id_token,
         this.getRedirectUri(),
-        request.state
+        reqState
```

Comment:
reqState could still be undefined right?

Reply:
Oh i see what you are trying to guard against, could do this

```javascript
request && request.state
```

Reply:
It can but we cover that use case if it isn't defined in ServerRequestParameters. The issue is when we are referencing undefined.value

Reply:
instead of the if block, the inline && will be safe for this.

Reply:
I would like to stick to one style, @pkanher617 made some changes already with the above code. Let me redo all the checks in one pattern and send out for the review, once I test it.

Reply:
okay consistency is fine, I just think the if check and set adds some mental overhead, and inline is a little more expressive, basically saying, we aren't sure this is set but if it is, grab state

Reply:
state can be undefined, but if request is undefined we get the error.

Reply:
so we don't want the request.state check in the if condition

Reply:
In the end it doesn't really matter, either way is functionally the same.